modules/cli/src/main/scala/scala/cli/commands/publish/Publish.scala

```diff
@@ -618,22 +652,25 @@ object Publish extends ScalaCommand[PublishOptions] {
           case Some("ivy2-local") =>
             ivy2Local
           case Some("central" | "maven-central" | "mvn-central") =>
-            centralRepo("https://oss.sonatype.org")
+            value(centralRepo("https://oss.sonatype.org"))
           case Some("central-s01" | "maven-central-s01" | "mvn-central-s01") =>
-            centralRepo("https://s01.oss.sonatype.org")
+            value(centralRepo("https://s01.oss.sonatype.org"))
           case Some(repoStr) =>
-            val repo0 = RepositoryParser.repositoryOpt(repoStr)
-              .collect {
-                case m: MavenRepository =>
-                  m
-              }
-              .getOrElse {
-                val url =
-                  if (repoStr.contains("://")) repoStr
-                  else os.Path(repoStr, Os.pwd).toNIO.toUri.toASCIIString
-                MavenRepository(url)
-              }
-              .withAuthentication(authOpt)
+            val repo0 = {
+              val r = RepositoryParser.repositoryOpt(repoStr)
+                .collect {
```

Comment:
why the newline? 
imo it's readable enough to contain the `collect` as a single line:
```scala
.collect { case m: MavenRepository => m }
```

Reply:
It's kind of arbitrary… I tend to always put `case` clauses on new lines, it's more readable this way IMO. I also don't put `{ … }` blocks on a single line for the same reason (only `( … )` ones).